rocketmq-spring-boot/src/main/java/org/apache/rocketmq/spring/support/RocketMQUtil.java

```diff
@@ -254,4 +254,20 @@ public static String getInstanceName(RPCHook rpcHook, String identify) {
             .append(separator).append(UtilAll.getPid());
         return instanceName.toString();
     }
+
+    public static String getNameServerString(List<String> nameServer) {
+        if (nameServer == null || nameServer.isEmpty()) {
+            throw new IllegalArgumentException("property nameServer is empty");
+        }
+        String nameServerToString;
+        if (nameServer.size() > 1) {
+            nameServerToString = String.join(";", nameServer);
+        } else {
+            nameServerToString = nameServer.get(0);
+            if (nameServerToString.contains(";")) {
+                log.warn("name-server format `host:port;host:port` is deprecated.");
```

Comment:
Should have a recommendation for a newer style.

Reply:
IMO that logic shows a separate property is warranted.